projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -168,6 +191,23 @@ export class ExplorerComponent {
       this.initialState$.pipe(map(value => value.contextToggle.value.dashboardContext)),
       this.contextChangeSubject
     );
+    this.featureStateResolver.getFeatureState(ApplicationFeature.SavedQueries).subscribe(featureState => {
+      this.enableSavedQueries = featureState === FeatureState.Enabled ? true : false;
+    });
+  }
+
+  public onClickSaveQuery(): void {
+    this.preferenceService.get('savedQueries', []).subscribe(queries => {
+      this.savedQueries = queries as SavedQuery[];
+    });
+
+    const currentScope =
+      this.filters[0]?.metadata.scope === SPAN_SCOPE ? ScopeQueryParam.Spans : ScopeQueryParam.EndpointTraces;
+    const currentFilters = this.filters.map(filter => filter.urlString);
```

Comment:
I think this block and the one that follows should be within the subscribe part? 
Also, are we gaining anywhere by maintaining `this.savedQueries` given that we are always getting `savedQueries` from `preferenceService` and setting it again. I think it can be a simple `const` within the scope of the handler.

Reply:
Moved the subscription to the constructor instead of onClick so we subscribe only once. And added a const instead to the onClick handler ✅